### PR TITLE
Upload Code Scanning Sarif results to GitHub

### PR DESCRIPTION
**Pull request by ganeshkumarsv**
Add a method to upload Code Scanning Sarif results to GitHub

### CODE REVIEW

File: github/code-scanning_test.go
Comment (ganeshkumarsv) [reply github/code-scanning_test.go]: Thanks for the review. changed to `cmp`

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:125]: Please delete this blank line so that the auto-generated Go docs get associated correctly with the struct.

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:122]: Let's please change the name of this to `SarifAnalysis` and add the 3 missing fields found in the documentation pointed to on line 124:

* `checkout_uri`
* `started_at`
* `tool_name`

The, let's please move this entire struct down to right before the new `UploadSarif` method and also use it as one of the parameters for uploading the analysis results.

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:192]: As discussed above, let's change this to:

```suggestion
func (s *CodeScanningService) UploadSarif(ctx context.Context, owner, repo string, sarif *SarifAnalysis) (*SarifID, *Response, error) {
```

Obviously, you will need to make a new `SarifID` struct to return the data that is documented in the GitHub v3 API docs URL above.

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:187]:
```suggestion
// For the parameter sarif, you must first compress your SARIF file using gzip and then translate the contents of the file into a Base64 encoding string.
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:199]: Please add a blank line before this line for consistency with the rest of the repo.

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:203]: Please add a blank line before this line for consistency with the rest of the repo.

Also, this is not correct... the `AnalysisResults` struct is not being passed back in the response.  Please re-read the documentation and create a new `SarifID` struct that has the `id` and `url` fields in it and return that instead.

File: github/code-scanning_test.go
Comment (gmlewis) [inline github/code-scanning_test.go:78]: Please add a blank line before this line for consistency with the rest of this repo.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @ganeshkumarsv , but as written, this code is not going to work.

Once you have addressed all my comments, please actually test out your code using the GitHub API and verify that it works correctly.

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:134]: Please add a godoc-style comment to describe this new struct.

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:130]:
```suggestion
	StartedAt   *Timestamp `json:"started_at,omitempty"`
```

File: github/code-scanning_test.go
Comment (gmlewis) [reply github/code-scanning_test.go:79]: Please add a blank line before this line (79) for consistency with the rest of this repo.

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Comment by ganeshkumarsv** [comment]
@gmlewis Thanks for all the review comments! I think I addressed all of them and tested. Please let me know if anything else needs to be changed.

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:134]:
```suggestion
// SarifID identifies a sarif analysis upload.
```

File: github/code-scanning_test.go
Comment (gmlewis) [reply github/code-scanning_test.go:79]: This comment has not yet been addressed.

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Comment by ganeshkumarsv** [comment]
addressed the remaining comments.

**Review by gmlewis — APPROVED** [review]
Thank you, @ganeshkumarsv !
LGTM.

Awaiting second LGTM (from any other go-github contributor) before merging.

**Comment by ganeshkumarsv** [comment]
@wesleimp @gmlewis 
gentle bump on the PR 🙏

**Review by viveknathani — APPROVED** [review]

**Comment by viveknathani** [comment]
> 
> 
> Thank you, @ganeshkumarsv ! LGTM.
> 
> Awaiting second LGTM (from any other go-github contributor) before merging.

LGTM!

**Comment by gmlewis** [comment]
Thank you, @viveknathani !
Merging.

**Comment by ganeshkumarsv** [comment]
@gmlewis can we have a new release for go-github? 🙇‍♂️

**Comment by gmlewis** [comment]
> @gmlewis can we have a new release for go-github? 🙇‍♂️

Yes, I'll work on that today.

**Comment by ganeshkumarsv** [comment]
Thank you! 😊 🙇‍♂️

**Comment by gmlewis** [comment]
@ganeshkumarsv - this has been released into `v40`: https://github.com/google/go-github/releases/tag/v40.0.0

**Pull request merged**
